notebooks/dia_kernel_exploration.ipynb

```diff
@@ -0,0 +1,1336 @@
+    "            A list of dicts having a \"source\" and \"footprint\"\n",
+    "            field for the Sources deemed to be appropriate for Psf\n",
+    "            matching. Can be the output from ``selectKernelSources``.\n",
+    "        preconvolved : `bool`, optional\n",
+    "            Was the science image convolved with its own PSF?\n",
+    "\n",
+    "        Returns\n",
+    "        -------\n",
+    "        results : `lsst.pipe.base.Struct`\n",
+    "\n",
+    "            ``psfMatchingKernel`` : `lsst.afw.math.LinearCombinationKernel`\n",
+    "                Spatially varying Psf-matching kernel.\n",
+    "            ``backgroundModel``  : `lsst.afw.math.Function2D`\n",
+    "                Spatially varying background-matching function.\n",
+    "        \"\"\"\n",
+    "        # Just debugging that we're really running this modified task\n",
+    "        self.log.info(\"Running Modified Make Kernel Task\")\n",
+    "\n",
+    "        kernelCellSet = self._buildCellSet(template.maskedImage, science.maskedImage, kernelSources)\n",
+    "        # Calling getPsfFwhm on template.psf fails on some rare occasions when\n",
+    "        # the template has no input exposures at the average position of the\n",
+    "        # stars. So we try getPsfFwhm first on template, and if that fails we\n",
+    "        # evaluate the PSF on a grid specified by fwhmExposure* fields.\n",
+    "        # To keep consistent definitions for PSF size on the template and\n",
+    "        # science images, we use the same method for both.\n",
+    "        try:\n",
+    "            templateFwhmPix = getPsfFwhm(template.psf)\n",
+    "            scienceFwhmPix = getPsfFwhm(science.psf)\n",
+    "        except InvalidParameterError:\n",
+    "            self.log.debug(\"Unable to evaluate PSF at the average position. \"\n",
+    "                           \"Evaluting PSF on a grid of points.\"\n",
+    "                           )\n",
+    "            templateFwhmPix = evaluateMeanPsfFwhm(template,\n",
+    "                                                  fwhmExposureBuffer=self.config.fwhmExposureBuffer,\n",
+    "                                                  fwhmExposureGrid=self.config.fwhmExposureGrid\n",
+    "                                                  )\n",
+    "            scienceFwhmPix = evaluateMeanPsfFwhm(science,\n",
+    "                                                 fwhmExposureBuffer=self.config.fwhmExposureBuffer,\n",
+    "                                                 fwhmExposureGrid=self.config.fwhmExposureGrid\n",
+    "                                                 )\n",
+    "\n",
+    "        if preconvolved:\n",
+    "            scienceFwhmPix *= np.sqrt(2)\n",
+    "            \n",
+    "        ### THESE LINES ARE PROBABLY WHERE YOU WANT TO CHANGE: BEGIN ###\n",
+    "        basisList = self.makeKernelBasisList(templateFwhmPix, scienceFwhmPix,\n",
```

Comment:
Modifying `basisList` could help us test the pipeline performance with different spatial degree of freedom, kernel bases (Gaussian-Polynomial, delta bases, modified delta bases), decorrelation methods, and spatial PSF model (power series, B-spline).